src/views/ClusterViz/index.js

```diff
@@ -0,0 +1,239 @@
+        if (modelId) {
+            const clusterDataRequest = new ProjectClusterDataRequest({
+                setProjectClusterData: this.props.setProjectClusterData,
+                setState: params => this.setState(params),
+            });
+            this.clusterDataRequest = clusterDataRequest.create(modelId, projectId);
+            this.clusterDataRequest.start();
+        }
+    }
+
+    stopRequestForClusterData = () => {
+        if (this.clusterDataRequest) {
+            this.clusterDataRequest.stop();
+        }
+    }
+
+    headers = [
+        {
+            key: 'value',
+            label: _ts('clusterViz', 'tableValueLabel'),
+            order: 1,
+            sortable: true,
+            comparator: (a, b) => compareString(a.value, b.value),
+        },
+        {
+            key: 'score',
+            label: _ts('clusterViz', 'tableScoreLabel'),
+            order: 2,
+            sortable: true,
+            comparator: (a, b) => compareNumber(a.score, b.score),
+        },
+    ];
+
+    render() {
+        const {
+            clusterDataPending,
+            createClusterPending,
+        } = this.state;
+
+        const {
+            className,
+            activeProject,
+            projectClusterData,
+        } = this.props;
+
+        const {
+            clusterTableExtractor,
+            headers,
+        } = this;
+
+        const clusters = projectClusterData.clusterData || [];
+
+        const clusterGroup = groupList(clusters, d => d.cluster);
+
+        const nodes = clusters.map(cluster => (
+            { id: cluster.value, group: cluster.cluster }
+        ));
+
+        const maxNodes = Object.entries(clusterGroup).map(values => (
+            { id: values[0], node: this.getMaxNode(values[1]) }
+        ));
+
+        const links = [].concat(...Object.entries(clusterGroup).map((values) => {
+            const target = maxNodes.find(node => node.id === values[0]);
+            const source = values[1].filter(node => target.node.value !== node.value);
+            return source.map(node => (
+                { source: node.value, target: target.node.value, value: 1 }
+            ));
+        }));
+
+        return (
+            <div className={`${styles.cluster} ${className}`}>
+                <header className={styles.header}>
+                    {_ts('clusterViz', 'clusterVizTitle')}
+                </header>
+                <div className={styles.container}>
+                    {
+                        (createClusterPending || clusterDataPending) &&
+                        <LoadingAnimation />
+                    }
+                    <ForceDirectedGraphView
+                        className={styles.forcedDirectedGraph}
+                        data={{ nodes, links }}
+                        idAccessor={d => d.id}
+                        useVoronoi={false}
+                        groupAccessor={d => d.group}
+                        valuAccessor={d => d.value}
```

Comment:
Use static declaration of accessor functions and use them here, removes the inefficiency.